MergerServiceUnitTests/Runners/TaskExecutorTest.cs

```diff
@@ -0,0 +1,156 @@
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Abstractions;
+using System.Linq;
+
+namespace MergerLogicUnitTests.Utils
+{
+  [TestClass]
+  [TestCategory("unit")]
+  [TestCategory("runners")]
+  public class TaskExecutorTest
+  {
+    #region mocks
+
+    private MockRepository _mockRepository;
+
+    private Mock<IDataFactory> _dataFactoryMock;
+    private Mock<ITimeUtils> _timeUtilsMock;
+    private Mock<IConfigurationManager> _configurationManagerMock;
+    private Mock<ILogger<TaskExecutor>> _taskExecutorLoggerMock;
+    private Mock<IMetricsProvider> _metricsProviderMock;
+    private Mock<ITaskUtils> _taskUtilsMock;
+    private Mock<ITileScaler> _tileScalerMock;
+    private Mock<ILogger<TileMerger>> _tileMergerLoggerMock;
+
+    private ActivitySource _testActivitySource;
+    private ITileMerger _testTileMerger;
+    private IFileSystem _testFileSystem;
+
+    #endregion
+
+    public static IEnumerable<object[]> GetSingleTileTestParameters()
+    {
+      yield return new object[] { 1, true };
+      yield return new object[] { 3, true };
+      yield return new object[] { 54, true };
+      yield return new object[] { 1, false };
+      yield return new object[] { 5, false };
+      yield return new object[] { 14, false };
+    }
+
+    [TestInitialize]
+    public void BeforeEach()
+    {
+      this._mockRepository = new MockRepository(MockBehavior.Loose);
+
+      this._dataFactoryMock = this._mockRepository.Create<IDataFactory>();
+      this._timeUtilsMock = this._mockRepository.Create<ITimeUtils>();
+      this._configurationManagerMock = this._mockRepository.Create<IConfigurationManager>();
+      this._taskExecutorLoggerMock = this._mockRepository.Create<ILogger<TaskExecutor>>();
+      this._metricsProviderMock = this._mockRepository.Create<IMetricsProvider>();
+      this._taskUtilsMock = this._mockRepository.Create<ITaskUtils>();
+      this._tileScalerMock = this._mockRepository.Create<ITileScaler>();
+      this._tileMergerLoggerMock = this._mockRepository.Create<ILogger<TileMerger>>();
+
+      this._testActivitySource = new ActivitySource("test");
+      this._testTileMerger = new TileMerger(_tileScalerMock.Object, _tileMergerLoggerMock.Object);
+      this._testFileSystem = new FileSystem();
+
+      this._configurationManagerMock.Setup(configManager => configManager.GetConfiguration<int>("GENERAL", "batchSize", "batchMaxSize")).Returns(1);
+      this._configurationManagerMock.Setup(configManager => configManager.GetConfiguration<bool>("GENERAL", "batchSize", "limitBatchSize")).Returns(true);
+      this._configurationManagerMock.Setup(configManager => configManager.GetConfiguration<long>("GENERAL", "batchMaxBytes")).Returns(1);
+    }
+
+    [TestMethod]
+    [DynamicData(nameof(GetSingleTileTestParameters), DynamicDataSourceType.Method)]
+    public void WhenGivenSourcesWithOneTile_ShouldWriteAllTilesToTarget(int numberOfSources, bool isTargetNew)
```

Comment:
don't we want to add a check for tile batch memory and count limits?

Reply:
added tests for it